### ISSUE TITLE
Table-style for creating Table of Contents

### ISSUE THREAD

**Issue by BrittanyBunk** [internetarchive/openlibrary#2384]
It's difficult to format the table of contents, so if people could just add a row or column that they need to and then fill in the cells, that would make it much easier to fill in and have it look nice.

**Comment by cdrini** [comment]
Hey @BrittanyBunk ! Thanks for the feedback. Please do try to follow the issue templates; it saves a lot of back and forth and makes issues easier to read for the maintainers, since they all look consistent.

If I'm following correctly, you're suggesting that the editing user interface include an editable table format? Like having an "Add a row" button, which will have the necessary columns, and which can then be filled in with the section details? Sort of like a spreadsheet?

In what ways do you think the current method is difficult?

**Comment by BrittanyBunk** [comment]
Yes, like a spreadsheet. It's difficult right now, because the formatting goes awry when I try to follow the directions. Also, it's tricky to copy/paste and have it come out right if I need to add the | <- just a hassle. Also, after adding it all in, I find that there's extra |'s added, and some of the ones I add get removed. So it just doesn't work and I just abandon it.

**Comment by mekarpeles** [comment]
Relates to #683

**Comment by guyjeangilles** [comment]
It sounds like we'd need to add functionality to insert data instead of editing the whole Table of Contents object. Assigning @hornc per slack conversations since it seems like something it'd spearhead.

**Comment by BrittanyBunk** [comment]
@guyjeangilles @mekarpeles thanks for moving this forward right now, as I feel it's needed more than other issues (that I've created) to move forward at this time :)

**Comment by BrittanyBunk** [comment]
To continue the progress, here are the names of the columns: #, Header, Description/Subtitle (optional), Page number. I'm not sure how to add tabbing into a spreadsheet, but I'm thinking that any tabbed header would have a symbol next to it - like ">" for each tab, like ">>>" = 3 tabs in the header category.

**Comment by mekarpeles** [comment]
Also see: https://github.com/internetarchive/openlibrary/issues/683

**Comment by cdrini** [comment]
This is still relevant; although I'm marking as blocked, since we do not want to create a new UI component, but re-use the UI component in tocky.
